app/code/Magento/QuoteConfigurableOptions/Model/Cart/BuyRequest/SuperAttributeDataProvider.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteConfigurableOptions\Model\Cart\BuyRequest;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Model\Cart\BuyRequest\BuyRequestDataProviderInterface;
+use Magento\Quote\Model\Cart\Data\CartItem;
+
+/**
+ * DataProvider for building super attribute options in buy requests
+ */
+class SuperAttributeDataProvider implements BuyRequestDataProviderInterface
+{
+    private const OPTION_TYPE = 'configurable';
+
+    /**
+     * @inheritdoc
+     *
+     * @throws LocalizedException
+     */
+    public function execute(CartItem $cartItem): array
+    {
+        $configurableProductData = [];
+        foreach ($cartItem->getSelectedOptions() as $optionData) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            $optionData = \explode('/', base64_decode($optionData->getId()));
+
+            if ($this->isProviderApplicable($optionData) === false) {
+                continue;
+            }
+            $this->validateInput($optionData);
+
+            [, $attributeId, $valueIndex] = $optionData;
+            $configurableProductData[$attributeId] = $valueIndex;
+        }
+
+        return ['super_attribute' => $configurableProductData];
+    }
+
+    /**
+     * Checks whether this provider is applicable for the current option
+     *
+     * @param array $optionData
+     * @return bool
+     */
+    private function isProviderApplicable(array $optionData): bool
+    {
+        if ($optionData[0] !== self::OPTION_TYPE) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Validates the provided options structure
+     *
+     * @param array $optionData
+     * @throws LocalizedException
+     */
+    private function validateInput(array $optionData): void
+    {
+        if (count($optionData) !== 3) {
```

Comment:
the mentioned structure is an implementation detail that should not be exposed to a client. the only reason we are using base64 because we can not introduce unique ID for all options values in a backward-compatible way right now. but we are going to introduce it eventually.


Anyway, all IDs should be retrieved from the regular option values query.